doc/source/ray-air/check-ingest.rst

```diff
@@ -3,444 +3,171 @@
+* Adjust the ``prefetch_batches`` argument for iter batches. This can be useful if bottlenecked on the network.
+* Finally, you can use ``print(ds.stats())`` or ``print(iterator.stats())`` to print detailed timing information about Ray Data performance.
 
-How do I pass in a :py:class:`~ray.data.dataset_pipeline.DatasetPipeline` to my ``Trainer``?
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-The Trainer interface only accepts a standard :py:class:`~ray.data.Dataset` and not a :py:class:`~ray.data.dataset_pipeline.DatasetPipeline`.
-Instead, you can configure the ingest via the ``dataset_config`` that is passed to your ``Trainer``. Internally, Ray AIR will
-convert the provided :py:class:`~ray.data.Dataset` into a :py:class:`~ray.data.dataset_pipeline.DatasetPipeline` with the specified configurations.
+Custom data config (advanced)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-See the :ref:`Enabling Streaming Ingest <air-streaming-ingest>` and :ref:`Shuffling Data <air-shuffle>` sections for full examples.
+For use cases not covered by the default config class, you can also fully customize exactly how your input datasets are splitted. To do this, you need to define a custom ``DataConfig`` class (DeveloperAPI). The ``DataConfig`` class is responsible for that shared setup and splitting of data across nodes.
 
-How do I shard validation and test datasets?
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+.. code:: python
 
-By default only the `"train"` Dataset is sharded. To also shard validation and test datasets, you can configure the ``dataset_config``
-that is passed to your ``Trainer``.
-See the :ref:`Splitting Auxiliary Datasets <air-splitting-aux-datasets>` section for a full example.
+    train_ds = ...   # As before.
+
+    # Note this example class is doing the same thing as
+    # the basic DataConfig impl included with Train.
+    class MyCustomDataConfig(train.DataConfig):
+       def get_shards(
+            self,
+            datasets: Dict[str, Dataset],
+            world_size: int,
+            worker_node_ids: List[NodeIdStr],
+            **kwargs):
+
+          # Configure Ray Data for ingest 
+          ctx = ray.data.DataContext.get_current()
+          ctx.execution_options = ctx.ingest_default_options()
+
+          # Split the stream into shards.
+          iterator_shards = datasets["train"].streaming_split(
+               world_size,
+               equal=True,
+               locality_hints=worker_node_ids)
+
+          # Create shards for each worker and return.
+          outputs = {}
+          for rank in range(world_size):
+              outputs[rank] = {"train": iterator_shards[rank]}
+          self.output_splits = outputs
```

Comment:
Nit, if this method is intended to be overridden by subclasses. Better make it side-effect free. I.e., just "return outputs", not changing "self.output_splits".